utils/self-extracting-executable/compressor.cpp

```diff
@@ -228,21 +239,36 @@ int saveMetaData(char* filenames[], int count, int output_fd, const MetaData& me
 }
 
 /// Fills metadata and calls compression function for each file
-int compressFiles(char* filenames[], int count, int output_fd, int level, const struct stat& info_out)
+int compressFiles(const char* out_name, const char* exec, char* filenames[], int count, int output_fd, int level, const struct stat& info_out)
 {
     MetaData metadata;
     size_t sum_file_size = 0;
-    metadata.number_of_files = htole64(count);
+    int is_exec = exec && *exec ? 1 : 0;
+    metadata.number_of_files = htole64(count + is_exec);
     off_t pointer = info_out.st_size;
 
+    uint64_t total_size = 0;
+    uint64_t total_compressed_size = 0;
+
     /// Store information about each file and compress it
-    FileData* files_data = new FileData[count];
-    char * names[count];
-    for (int i = 0; i < count; ++i)
+    FileData* files_data = new FileData[count + is_exec];
+    const char * names[count + is_exec];
+    for (int i = 0; i <= count; ++i)
     {
-        printf("Compressing: %s\n", filenames[i]);
+        const char* filename = nullptr;
+        if (i == count)
+        {
+            if (!is_exec)
+                continue;
+            filename = exec;
+            files_data[i].exec = true;
+        }
```

Comment:
Consider extracting the loop body into function and make one explicit extra call.

Reply:
this will involve too many arguments making it even more obscure
let's leave it as is...